docs/antrea-proxy.md

```diff
@@ -0,0 +1,218 @@
+will be load-balanced directly by AntreaProxy to a CoreDNS Pod endpoint. This
+means that NodeLocal DNSCache is completely bypassed, which is probably not
+acceptable for users who want to leverage this feature to improve DNS
+performance in their clusters. While these users can update the Pod
+configuration to use the local IP assigned by NodeLocal DNSCache to the "dummy"
+interface, this is not always ideal in the context of CaaS, as it can require
+everyone running Pods in the cluster to be aware of the situation.
+
+This is the reason why we initially introduced the `skipServices` configuration
+option for AntreaProxy in Antrea v1.4. By adding the kube-dns Service (which
+exposes CoreDNS) to the list, you can ensure that AntreaProxy will "ignore" Pod
+DNS queries, and that they will be forwarded to NodeLocal DNSCache. You can edit
+the `antrea-config` ConfigMap as follows:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    antreaProxy:
+      skipServices: ["kube-system/kube-dns"]
+```
+
+### When you want your external LoadBalancer to handle Pod traffic
+
+In some cases, the external LoadBalancer for a cluster provides additional
+capabilities (e.g., TLS termination) and it is desirable for Pods to access
+in-cluster Services through the external LoadBalancer. By default, this is not
+the case as both kube-proxy and AntreaProxy will install rules to load-balance
+this traffic directly at the source Node (even when the destination IP is set to
+the external `loadBalancerIP`). To circumvent this behavior, we introduced the
+`proxyLoadBalancerIPs` configuration option for AntreaProxy in Antrea v1.5. This
+option defaults to `true`, but when setting it to `false`, AntreaProxy will no
+longer load-balance traffic destined to external `loadBalancerIP`s, hence
+ensuring that this traffic can go to the external LoadBalancer. You can set it
+to `false` by editing the `antrea-config` ConfigMap as follows:
+
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    antreaProxy:
+      proxyLoadBalancerIPs: false
+```
+
+There are two important prerequisites for this feature:
+
+* You must enabled `proxyAll` and [remove kube-proxy](#removing-kube-proxy) from
+  the cluster, otherwise kube-proxy will still load-balance the traffic and you
+  will not achieve the desired behavior.
+* Your external LoadBalancer must SNAT the traffic, in order for the reply
+  traffic to go back through the external LoadBalancer.
+
+## Known issues or limitations
+
+* Due to some restrictions on the implementation of Services in Antrea, the
+  maximum number of Endpoints that Antrea can support at the moment is 800. If
+  the number of Endpoints for a given Service exceeds 800, extra Endpoints will
+  be dropped. This will be fixed eventually.
```

Comment:
be dropped (remote endpoints are preferentially dropped). This will be fixed eventually.